src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -631,6 +643,65 @@ public void addEnvVars(List<TemplateEnvVar> envVars) {
         }
     }
 
+    /**
+     * Deletes the cloud.
+     */
+    @RequirePOST
+    public HttpResponse doDoDelete(StaplerRequest req, StaplerResponse rsp) throws IOException {
```

Comment:
we may want to push this as well to the parent

Reply:
simply because we do not have to do this context matching which feels like a hack

Reply:
Add `@AncestorInPath KubernetesCloud cloud` to the method signature, should look up what you expect.

Reply:
An alternative would be to store a transient back reference to the owning cloud inside of the `PodTemplate` object.

Reply:
Using the annotation is not allowed, however, the following is the same: `KubernetesCloud kubernetesCloud = req.findAncestorObject(KubernetesCloud.class);`

Reply:
> the annotation is not allowed

Maybe you are just not doing it right. Try

```java
public HttpResponse doDoDelete(@AncestorInPath KubernetesCloud cloud) throws IOException {
```

You do not need the `req` or `rsp` parameters.